File: 03-array/07-reverse-an-array/07-reverse-an-array.ts
const reverse_an_array = (nums: number[]): number[] => {
  let lp = 0
  let rp = nums.length - 1
  while (lp < rp) {
    const temp = nums[lp]
    nums[lp] = nums[rp]
    nums[rp] = temp
    lp += 1
    rp -= 1
  }
  return nums
}

console.log(reverse_an_array([10, 20, 30]))

console.log(reverse_an_array([10, 20, 30, 40, 50]))

File: 03-array/07-reverse-an-array/07-reverse-an-array.rb
def reverse_an_array(nums)
  lp, rp = 0, nums.length-1
  while lp < rp
    nums[lp], nums[rp] = nums[rp], nums[lp]
    lp, rp = lp+1, rp-1
  end
  return nums
end

res = reverse_an_array([10, 5, 7, 30])
print(res)

res = reverse_an_array([30, 20, 5])
print(res)
